dynamo/prediction/fate.py

```diff
@@ -27,8 +27,8 @@ def fate(
     init_states: Optional[np.ndarray] = None,
     basis: Optional[None] = None,
     layer: str = "X",
-    dims: Union[tuple([list, None] + list(np.ScalarType))] = None,
-    genes: Union[list, None] = None,
+    dims: Union[np.ScalarType, Optional[List[np.ScalarType]]] = None,
```

Comment:
I think `np.ScalarType` is not a 'type' object itself, but a tuple of 'type' objects. The type hint of `dims` may not work as expected.